File: course_1_1/Task_5.py
class Queue:
    def __init__(self):
        self.queue = []

    def enqueue(self, item):  # O(1)
        self.queue.append(item)

    def dequeue(self):  # O(n)
        if not self.size():
            return None
        queue_start = self.queue[0]
        del self.queue[0]
        return queue_start

    def size(self):
        return len(self.queue)

File: course_1_11_java_ads_refresh_practice/Task_5.java
import java.util.*;

public class Queue<T> {

    ArrayList<T> storage;

    public Queue() {
        // initializing the queue's internal storage
        storage = new ArrayList<>();

    }

    // Task number: 5.1.1
    // Short description: Implement method of getting size of queue
    // Time complexity: O(1)
    // Space Complexity: O(1)
    public void enqueue(T item) {
        storage.addLast(item);
    }

    // Task number: 5.1.2
    // Short description: Implement method of putting value in queue
    // Time complexity: O(N)
    // Space Complexity: O(1)
    public T dequeue() {
        if (size() == 0) {
            return null; // null if queue is already empty
        }
        return storage.removeFirst();
    }

    // Task number: 5.1.3
    // Short description: Implement method of getting value from queue
    // Time complexity: O(1)
    // Space Complexity: O(1)
    public int size() {
        return storage.size();
    }

}

